src/aggregate/aggregate_exec.c

```diff
@@ -1056,6 +1060,9 @@ int RSCursorCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 
   } else if (cmdc == 'D') {
     int rc = Cursors_Purge(&g_CursorsList, cid);
+    if(rc != REDISMODULE_OK) {
```

Comment:
Can we avoid failed attempts to search the "wrong" list? (also with `Cursors_TakeForExecution`)
For example, if `CursorList_GenerateId` would return odd ids for coordinator cursors, and even ids for user cursors - we could easily detect which list to look for.